contracts/upgradeable-Bridge/FiberRouter.sol

```diff
@@ -301,74 +341,57 @@ contract FiberRouter is Ownable, TokenReceivable {
        will also be required on the other end
      @param crossTargetAddress The target address for the swap
+     @param gasFee The gasFee being charged on withdrawal
      */
-    function nonEvmSwapAndCrossOneInch(
-        uint256 amountIn,
-        uint256 amountOut, // amountOut on oneInch
-        string memory crossTargetNetwork, //cudos-1
-        string memory crossTargetToken, //acudos
-        string memory crossTargetAddress, //acudosXYZ
-        bytes memory oneInchData,
-        address fromToken,
-        address foundryToken,
-        bytes32 withdrawalData
-    ) external nonReentrant {
-        // Validation checks
-        require(fromToken != address(0), "From token address cannot be zero");
-        require(
-            foundryToken != address(0),
-            "Foundry token address cannot be zero"
-        );
-        require(amountIn != 0, "Amount in must be greater than zero");
-        require(
-            amountOut != 0,
-            "Amount cross minimum must be greater than zero"
-        );
-        require(bytes(oneInchData).length != 0, "1inch data cannot be empty");
-        require(
-            bytes(crossTargetNetwork).length != 0,
-            "Cross target network cannot be empty"
-        );
-        require(
-            bytes(crossTargetToken).length != 0,
-            "Cross target token cannot be empty"
-        );
-        require(
-            bytes(crossTargetAddress).length != 0,
-            "Cross target address cannot be empty"
-        );
-        require(
-            withdrawalData != 0,
-            "FR: withdraw data cannot be empty"
-        );
-        amountIn = SafeAmount.safeTransferFrom(
-            fromToken,
-            _msgSender(),
-            address(this),
-            amountIn
-        );
-        uint256 settledAmount = _nonEvmSwapAndCrossOneInch(
-            amountIn,
-            amountOut,
-            crossTargetNetwork,
-            crossTargetToken,
-            crossTargetAddress,
-            oneInchData,
-            fromToken,
-            foundryToken
-        );
-        emit NonEvmSwap(
-            fromToken,
-            crossTargetToken,
-            block.chainid,
-            crossTargetNetwork,
-            amountIn,
-            _msgSender(),
-            crossTargetAddress,
-            settledAmount,
-            withdrawalData
-        );
-    }
+ function swapAndCrossOneInchETH(
+    uint256 amountOut, // amountOut on oneInch
+    uint256 crossTargetNetwork,
+    address crossTargetToken,
+    address crossTargetAddress,
+    bytes memory oneInchData,
+    address foundryToken,
+    bytes32 withdrawalData,
+    uint256 gasFee
+) external payable {
+    uint256 amountIn = msg.value - gasFee;
+    // Validation checks
+    require(amountIn != 0, "FR: Amount in must be greater than zero");
+    require(gasFee != 0, "FR: Gas fee must be greater than zero");
+    require(msg.value == amountIn + gasFee, "FR: msg.value must equal amountIn plus gasFee");
```

Comment:
Redundant

Reply:
the deployer pattern is already implemented and a PR is also created by @SibghatUllah1997 https://github.com/ferrumnet/MultiSwap/pull/39